Rlabkey/R/labkey.defaults.R

```diff
@@ -209,19 +207,52 @@ labkey.setDebugMode <- function(debug=FALSE)
     .lkdefaults[["debug"]] = debug;
 }
 
-handleError <- function(response, status_code, haltOnError) {
+isRequestError <- function(response, status_code) 
+{
+  status_code <- getStatusCode(response)
+  
+  return(status_code==500 | status_code >= 400)
+}
+
+getStatusCode <- function(response) 
+{
+  ## Error checking, decode data and return
+  status_code <- response$status_code
+  
+  #test for the situations where the header reports 200, but the JSON contains the error:
+  if (sum(grepl('application/json', response$headers[['content-type']])) > 0) {
+    c <- content(response, type = "application/json")
+    if (!is.null(c[['status']])) {
+      status_code <- c$status
+    }
+  }
+
+  return(status_code)  
+}
+
+handleError <- function(response, haltOnError) 
+{
   status <- http_status(response)
+  message = status$message
   
   ## pull out the error message if possible
   error <- content(response, type = "application/json")
-  message = status$message
   if (!is.null(error$exception))
   {
     message <- error$exception
   }
-  if (haltOnError)
+  if (haltOnError) {
+    status_code <- getStatusCode(response)
+
+    # Note: is this request was writing to a file, the error message JSON will be written to that file, so we delete.  
+    if (inherits(response$content, 'path')) {
+      if (file.exists(response$content)){
+        unlink(response$content)  
```

Comment:
Will this delete an existing file when `overwrite=TRUE` and the request has an error?

Reply:
I suppose, but it would already have been overwritten with the JSON content of this response.

The basic problem is that our WebDav API returns 200 status codes for many types of failures, and then includes the true status/exception in the JSON body.  As this currently works in R/httr, when I make these requests we're asking R to write the response to a file.  Under normal operation, that response is the actual desired file.  We dont generally want to write this whole file to memory.  For the code you point out to be called, that request would have already happened, and the result of the response written to that file.  

Also - I believe what's checked in fixes all your requests except 2:

- If labkey.get() is made against a directory, the server returns the HTML/JS code of the FileBrowser.  That's just how the WebDav controller works.  I am trying to think of ways around this, but dont have one right now.

- I'll have a fix in a minute; however, your issue with downloadFolder() has to do w/ not handling spaces in the folderPath properly, I think.  That's why it worked as expected for me and my example (webDavTest.R), but not yours.  I should have a fix on this soon.

Reply:
@labkey-tchad - I think the current version of this should fix the problems you noted.